src/main/kotlin/org/rust/ide/refactoring/inlineValue/InlineValueHandler.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.inlineValue
+
+import com.intellij.lang.Language
+import com.intellij.lang.refactoring.InlineActionHandler
+import com.intellij.openapi.command.WriteCommandAction.runWriteCommandAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.refactoring.RefactoringBundle
+import com.intellij.refactoring.util.CommonRefactoringUtil
+import org.rust.ide.refactoring.findBinding
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.psi.RsConstant
+import org.rust.lang.core.psi.RsLetDecl
+import org.rust.lang.core.psi.RsPatIdent
+import org.rust.lang.core.psi.ext.ancestorOrSelf
+
+class InlineValueHandler : InlineActionHandler() {
+    override fun isEnabledForLanguage(language: Language): Boolean = language is RsLanguage
+    override fun inlineElement(project: Project, editor: Editor, element: PsiElement) {
+        val context = getContext(project, editor, element) ?: return
+        val references = ReferencesSearch.search(context.binding).toList()
+        if (references.isEmpty()) {
+            showErrorHint(project, editor, "${context.name} is never used")
+            return
+        }
+
+        runWriteCommandAction(project) {
+            references.forEach {
+                it.element.replace(context.expr)
+            }
+            context.element.delete()
+        }
+    }
+
+    override fun canInlineElement(element: PsiElement): Boolean {
+        if (element.ancestorOrSelf<RsConstant>() != null) return true
```

Comment:
I wonder why Java's implementations like `InlineLocalHandler`, `InlineConstantFieldHandler`, `InlineMethodHandler` just check the type of `element` without checking ancestors

Reply:
If I just check for the type directly, it is not available for the whole constant/let declaration, which seems unintuitive for me:
```rust
const CONST: u32 = 5/*caret*/; // not available
const CONST: u32/*caret*/ = 5; // not available
const CONST:/*caret*/ u32 = 5; // not available
const CONST/*caret*/: u32 = 5; // available
/*caret*/const CONST: u32 = 5; // not available
```

Reply:
Actually to be on par with Java, we also need to check if the element is a reference to a constant/let decl, for this to work:
```rust
const CONST: u32 = 1;
let a = CONST/*caret*/ + 2;
```

Reply:
Clear! But this is exactly how it works in Java. The refactoring is available only when the caret is placed near a constant name or a reference to it.

Reply:
Okay, so should I change it to behave the same as Java/Kotlin?

Reply:
> Actually to be on par with Java, we also need to check if the element is a reference to a constant/let decl, for this to work

This works automagically right now =)

> so should I change it to behave the same as Java/Kotlin?

I think, yes